Building from source on Ubuntu 20.04 failed

I've been working on a environment that needs to run on **Ubuntu 20.04**, with ros2 **Humble**. I'm trying to build `ros2_controllers`, then I realized that it was missing the build of the dependencies such as `generate_parameter_library`. Since in this case we are using a unofficial ubuntu version for Humble, I'm build from source most of the dependencies. Until this moment what is working:

1. Installed `fmt`: apt install libfmt-dev
2. Cloned [PickNikRobotics/cpp_polyfills](https://github.com/PickNikRobotics/cpp_polyfills.git)
3. colcon build --packages-select tcb_span tl_expected
4. Cloned [PickNikRobotics/RSL](https://github.com/PickNikRobotics/RSL.git)
5. Unfortunately, in this one it was necessary to set the cmake_minimum_required to 3.16 at the CMakeList. The build worked, but I don't know how wrong this can be
6. Installed `typeguard`: $pip3 install typeguard
7. Cloned `generate_parameter_library`
8. Builded generate_parameter_library_py, parameter_traits, and generate_parameter_library using colcon
9. The error comes when building generate_parameter_library_example. The building fail with the following message:

```
CMake Error at /home/<username>/ros2_ws/install/generate_parameter_library/share/generate_parameter_library/cmake/generate_parameter_library.cmake:42 (cmake_path):
  Unknown CMake command "cmake_path".
Call stack (most recent call first):
  CMakeList.txt:15 (generate_parameter_library)
```

More datails about my setup:
- Python 3.8.10
- cmake version 3.16.3

--- a/generate_parameter_library/CMakeLists.txt
+++ b/generate_parameter_library/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.16)
+cmake_minimum_required(VERSION 3.20)
 project(generate_parameter_library NONE)
 
 # find dependencies

